graylog2-web-interface/src/components/login/LoginForm.tsx

```diff
@@ -39,54 +40,40 @@ const SigninButton = styled(ModalSubmit)(({ theme }) => css`
   }
 `);
 
-const LoginForm = ({ onErrorChange }: Props) => {
-  const [isLoading, setIsLoading] = useState(false);
-  let promise;
-  let usernameInput;
-  let passwordInput;
+const performLogin = ([username, password, host]: [string, string, string]) => SessionActions.login(username, password, host);
 
-  useEffect(() => () => {
-    if (promise) {
-      promise.cancel();
-    }
-  }, [promise]);
+const LoginForm = ({ onErrorChange }: Props) => {
+  const { mutateAsync: login, isLoading } = useMutation(performLogin,
```

Comment:
To keep the code cleaner, we can create the hook `useLoginMutation`  and move all logic there 

usage in LoginForm would look like 
`const { login, isLoading } = useLoginMutation()`

Reply:
:+1: